Upload an Exploratory PCA Notebook

rdvelazquez opened the pull request:
@dhimmel : Thought you might be interested in this as a reference or maybe adding some of it to the cognoma machine-learning / explore directory.

rdvelazquez committed:
Upload an Exploratory PCA Notebook

@dhimmel : Thought you might be interested in this as a reference or maybe adding some of it to the cognoma machine-learning / explore directory.

dhimmel reviewed (commented):
Excited to review.

Let's put this in a directory, e.g. `explore/pca-visualization`. Also please export notebook to a script for github reviewing.

rdvelazquez committed:
Delete PCA_Visualization.ipynb

rdvelazquez committed:
Create pca-visualization

rdvelazquez committed:
Delete pca-visualization

rdvelazquez committed:
Create README.md

rdvelazquez committed:
Add files via upload

rdvelazquez commented:
OK. I think I've now moved the file to an `exlore/pca-visualization` directory and exported the notebook to a script. 

Sorry about the multiple commits; I'm new to github. 

I didn't know of a way to move the file to a new folder (I have a Windows computer and I'm just using the github interface, not a command line) so I deleted the file and re-uploaded it into the new directory. I also added `'..' , '..',`  to the paths for the downloaded files, to account for the change in location.

Hopefully that worked! (fingers crossed)

dhimmel commented on explore/pca-visualization/PCA_Visualization.py:
```diff
@@ -0,0 +1,271 @@
+
+# In[6]:
+
+PlotExplainedVariance(pca, {'All Componetns':[0,TotalComponents]})
```
Spelling typo

dhimmel reviewed (commented):
@rdvelazquez amazing work! The variance explained plots will be really helpful. And the colored PCA scatterplots are also beautiful! @gwaygenomics take a look.

So it appears that gender doesn't have a huge impact on expression, whereas disease type does.

dhimmel commented:
@rdvelazquez I noticed you had a TODO comment. Let me know when you're finished on your end.

Also if you'll be at the meetup tonight, it would be great to have you briefly go over these findings (only if you want).

gwaybio commented on explore/pca-visualization/PCA_Visualization.py:
```diff
@@ -0,0 +1,271 @@
+        del neutralDiseaseDict[disease]
+neutralDiseaseList = list(neutralDiseaseDict.keys())
+# Print the list for reference
+print(neutralDiseaseList)
```
I would include `BRCA` and `TGCT` here as well. `BRCA` is a good control as there are _some_ male BRCA samples in the dataset. (although there are stronger signals in the data that should be explained by top PCs). `TGCT` is also an interesting case since some men with the disease have loss of Y chromosome expression.

rdvelazquez replied:
A webapp with visualization methods sounds intriguing!

I'll include `BRCA` when I address the other comments. Any preference on if I should just add it to the list or plot as it's own cell?

`TGCT` would be a little trickier (I assume it's only males). It might be interesting to see the PCs plotted with the total Y chromosome expression as point size or gray-scale but I think this is beyond the scope of this notebook.

gwaybio reviewed (commented):
Nice visualization and would be great to include in this repo! 

I have been thinking a lot about different visualization methods for this data built into a webapp (possibly outside scope of cognoma?) that would provide interactive tools to explore reduced dimension space according to different clinical and genetic parameters - so its good others are also thinking along these lines!

rdvelazquez commented:
Thanks for the comments!

> I noticed you had a TODO comment. Let me know when you're finished on your end.

I'm done. That was just an artifact. I'll remove the "TODO" comment when I address your other comments.

> Also if you'll be at the meetup tonight, it would be great to have you briefly go over these findings (only if you want).

I'll be a little late tonight but I'd be glad to briefly go over this.

See you tonight.

Commit:
Fix typo and add BRCA

Commit:
Remove TODO reference

rdvelazquez commented:
@dhimmel let me know if you have any other edits or comments on this PR.

dhimmel reviewed (approved):
Thanks for this stellar contribution!
